File: web-dashboard/remove_bg.js
const Jimp = require('jimp');
const fs = require('fs');
const path = require('path');

const spritesDir = path.join(__dirname, 'public', 'sprites');
const files = fs.readdirSync(spritesDir).filter(f => f.endsWith('.png') && !f.includes('_transparent'));

async function processImages() {
    for (const file of files) {
        const filePath = path.join(spritesDir, file);
        try {
            const image = await Jimp.read(filePath);

            // Tolerance for "white"
            const tolerance = 5;

            image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
                const red = this.bitmap.data[idx];
                const green = this.bitmap.data[idx + 1];
                const blue = this.bitmap.data[idx + 2];

                // If the pixel is very close to white
                if (red > 255 - tolerance && green > 255 - tolerance && blue > 255 - tolerance) {
                    this.bitmap.data[idx + 3] = 0; // set alpha to 0
                }
            });

            await image.writeAsync(filePath);
            console.log(`Processed ${file}`);
        } catch (e) {
            console.error(`Error processing ${file}:`, e);
        }
    }
}

processImages();

File: web-dashboard/remove_bg.py
import os
from PIL import Image

# Directories containing the images
directories = [
    '/Users/alan_wang/project-golem/web-dashboard/public/characters',
    '/Users/alan_wang/project-golem/web-dashboard/public/props',
    '/Users/alan_wang/project-golem/web-dashboard/public/office-assets/tech'
]

def remove_white_background(img_path):
    print(f"Processing: {img_path}")
    try:
        img = Image.open(img_path)
        img = img.convert("RGBA")
        
        datas = img.getdata()
        newData = []
        
        # Define a threshold for what constitutes "white" to handle slight anti-aliasing in AI generation
        # A higher threshold (e.g., 250) is safer for preserving light-colored details.
        threshold = 245
        
        for item in datas:
            # Check if pixel is close to white (R, G, B > threshold)
            # item[3] is the alpha channel, only process opaque pixels
            if item[3] > 0 and item[0] > threshold and item[1] > threshold and item[2] > threshold:
                # Change all white pixels to transparent
                newData.append((255, 255, 255, 0))
            else:
                newData.append(item)
                
        img.putdata(newData)
        img.save(img_path, "PNG")
        print(f"Saved: {img_path}")
    except Exception as e:
        print(f"Error processing {img_path}: {e}")

# Process all PNGs in the specified directories
for directory in directories:
    if not os.path.exists(directory):
        print(f"Directory not found: {directory}")
        continue
        
    print(f"\nScanning directory: {directory}")
    for filename in os.listdir(directory):
        if filename.endswith(".png"):
            file_path = os.path.join(directory, filename)
            remove_white_background(file_path)

print("\nBackground removal complete.")
